MHER-53: Debian packaging scripts - initial commit

BMartinos opened the pull request:
Added scripts to allow for the mHero mediator to be packages in a debian file

Commit:
MHER-53: Debian packaging scripts - initial commit
Added scripts to allow for the mHero mediator to be packages in a debian file

Commit:
MHER-53: Added control dependencies
Added openhim-core, nvm dependencies to the install script

BMartinos commented:
Hey @rcrichton Ive gotten quite far with the deb packaging but running into a weird error which im not sure what the problem is. In the control file ive added two dependencies and even though they are installed, the package installation fails because it does not find the dependencies. Have you experienced something like this before?

rcrichton commented on targets/trusty/debian/control:
```diff
@@ -0,0 +1,15 @@
+
+Package: openhim-mediator-mhero
+Architecture: amd64
+Depends: openhim-core, nvm
```
I think this actually need to be `openhim-core-js`. Also I'm not sure if there is a `nvm` package in the ubuntu repositories. Either way, this should be installed by the `openhim-core-js` package I think.

Commit:
MHER-53: Fixed required dependency
Fixed an incorrect dependency in the control file

Commit:
MHER-53: added additional install commands in postinst
I had to add a script to update node to version 4 for the mediator to run

BMartinos commented:
Hey @rcrichton I have added the necessary scripts and was able to successfully install in a clean vagrant instance. If you can review this for me when you get a chance. Thanks

rcrichton commented:
@BMartinos I'm just starting to review this now, but one quick first comment. We usually add all packaging file and folders to a `packaging` folder. Would you be able to do  that here to? I just keep it separate out from the source code.

rcrichton commented on targets/trusty/debian/changelog:
```diff
@@ -0,0 +1,95 @@
+
+ -- Martin Brocker <martin@jembi.org>  Fri, 29 Apr 2016 15:33:24 +0200
+
+openhim-mediator-mhero (0.1.0-1~trusty) trusty; urgency=low
```
We need to reset this back to the first entry as most of these builds are just testing builds. We should only commit the changelog when we upload the package.

rcrichton commented on targets/trusty/usr/share/openhim-mediator-mhero:
```diff
@@ -0,0 +1 @@
+Subproject commit 1420fd4cb809e0f1fcdf6f20d0b8a3a6c79fcee6
```
What is this?

rcrichton commented:
Perhaps also add a brief readme text in the packaging folder to tell someone how to go about the packaging.

rcrichton commented on cp-mediators-into-pkg.sh:
```diff
@@ -0,0 +1,15 @@
+#!/bin/bash
+set -e
+
+SRCDIR="/home/martin/git"
```
This isn't used. (May have been un-used in my script to)

rcrichton commented:
@BMartinos when I install this it seems to work well great however I'm having a problem with the mediator config.

It gets set to `"apiURL": "localhost:8080"` where it should be `"apiURL": "https://localhost:8080"`. I could be doing something wrong.

rcrichton committed:
Fix openinfoman URL

Commit:
MHER-53: Minot fixes - Moved packing to correct folder structure - Readme update
Had a few minor fixes
Moved packing into 'packaging' folder
Updated the readme to explain how to run the packaging script

Commit:
Merge branch 'master' of https://github.com/jembi/openhim-mediator-mhero into debian-packaging

Commit:
MHER-53: Updated .gitignore to not include packaging/builds

Commit:
Removed files in old structure

Commit:
Removed build package which was committed

BMartinos commented:
Hi @rcrichton I have made the corrections you mentioned. I not sure what that subproject commit was on the main repository. I think this might have been auto generated when I created my branch. If you can review it again for me please. Thanks

rcrichton committed:
Adds a Vagrantfile and misc changes

Fix changelog and submodule issue.
Add to gitignore file.
Reword script question.

rcrichton commented:
Thanks @BMartinos look good. I've added a few things to this including the Vagrantfile I used for testing.
